### ISSUE TITLE
`cv::cuda::pow` cannot handle negative values properly

### ISSUE THREAD

**Issue by cv3d** [opencv/opencv#12281]
##### System information (version)
- OpenCV => 4.0.0-pre
- Operating System / Platform => Windows 64 Bit (perhaps others)
- Compiler => Visual Studio 2013  (perhaps others)
- WITH_CUDA

##### Detailed description
Results of `cv::cuda::pow` with negative input and odd powers (e.g. 3) are all zeros (in case of int-type input) or positive value (in case of float-type input).

##### Steps to reproduce
```.py
npMat = np.arange(-2, 3) #[-2 -1  0  1  2]
cuMat = cv.cuda_GpuMat()
cuMat.upload(npMat)

print(npMat)
print(npMat**3)
print(cv.cuda.pow(cuMat, 3).download().ravel())
```
Note that I am using Python wrapper from #12234, but the bug should occur in C++ as well

##### Expected result
```
[-2 -1  0  1  2]
[-8 -1  0  1  8]
[-8 -1  0  1  8]
```

##### Actual result
```
[-2 -1  0  1  2]
[-8 -1  0  1  8]
[0 0 0 1 8] #for int64, int32, int16, int8 input types
#OR
[ 8.  1.  0.  1.  8.] #for float32, float64 input types
```

**Comment by alalek** [comment]
Input, actual result, expected result?

**Comment by cv3d** [comment]
@alalek I clarified the problem further.

**Comment by alalek** [comment]
CUDA itself doesn't like negative values for `__powf` function:
- https://devtalk.nvidia.com/default/topic/406489/unsuspected-work-of-pow-function-pow-device-function-works-incorrectly-witn-negative-numbers/
- https://devtalk.nvidia.com/default/topic/528493/__powf-x-y-gives-nan/


In OpenCV CUDA code I see [this hack](https://github.com/opencv/opencv/blob/3.4.2/modules/cudaarithm/src/cuda/math.cu#L296-L297), but I have no idea where it can zero result values.

What is uploaded `GpuMat` type?

**Comment by cv3d** [comment]
I tried various types (float32, float64, int64, int32, int16, int8), and all of them fail. Please check the comment in fron of the actual result for details.

**Comment by cv3d** [comment]
Well, there is a glitch in the hack!! The input needs to be passed as an absolute value to the `__pow` function, but it was passed as-is, and the `absf`-value was taken before the call to the pow function.

BTW: Please remove the `incomplete` tag

**Comment by cv3d** [comment]
I am closing this issue as the bot seems confused, and cannot recognize it is already fixed.

**Comment by alalek** [comment]
GitHub closes when patch is merged into master branch (after next "Merge 3.4" patch).
"Closing" keywords doesn't work for non-default branch.
